src/http_frontend/http_graph_executor_impl.cpp

```diff
@@ -0,0 +1,126 @@
+#pragma GCC diagnostic ignored "-Wunused-but-set-variable"
+#include "tensorflow_serving/util/net_http/server/public/server_request_interface.h"
+#pragma GCC diagnostic pop
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wdeprecated-declarations"
+#include "mediapipe/framework/calculator_graph.h"
+#include "mediapipe/framework/formats/image_frame.h"
+#include "mediapipe/framework/formats/image_frame_opencv.h"
+#pragma GCC diagnostic pop
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wall"
+#include "mediapipe/framework/formats/tensor.h"
+#include "mediapipe/framework/port/status.h"
+#pragma GCC diagnostic pop
+
+#include "../mediapipe_internal/mediapipe_utils.hpp"
+
+namespace ovms {
+
+Status deserializeInputSidePacketsFromFirstRequestImpl(
+    std::map<std::string, mediapipe::Packet>& inputSidePackets,  // out
+    const std::string& request) {                                // in
+    return StatusCode::OK;
+}
+
+const std::string& getRequestId(
+    const std::string& request) {
+    static const std::string _default{""};
+    return _default;
+}
+
+Status onPacketReadySerializeAndSendImpl(
+    const std::string& requestId,
+    const std::string& endpointName,
+    const std::string& endpointVersion,
+    const std::string& packetName,
+    const mediapipe_packet_type_enum packetType,
+    const ::mediapipe::Packet& packet,
+    HttpWriter& writer) {
+    std::string out;
+    OVMS_RETURN_ON_FAIL(
+        onPacketReadySerializeImpl(
+            requestId,
+            endpointName,
+            endpointVersion,
+            packetName,
+            packetType,
+            packet,
+            out));
+    writer.WriteResponseString(out);
+    writer.PartialReply();
+    return StatusCode::OK;
+}
+
+Status onPacketReadySerializeImpl(
+    const std::string& requestId,
+    const std::string& endpointName,
+    const std::string& endpointVersion,
+    const std::string& packetName,
+    const mediapipe_packet_type_enum packetType,
+    const ::mediapipe::Packet& packet,
+    std::string& response) {
+    response = packet.Get<std::string>();
+    return StatusCode::OK;
+}
+
+Status createAndPushPacketsImpl(
+    std::shared_ptr<const std::string> request,
+    stream_types_mapping_t& inputTypes,
+    PythonBackend* pythonBackend,
+    ::mediapipe::CalculatorGraph& graph,
+    ::mediapipe::Timestamp& currentTimestamp,
+    size_t& numberOfPacketsCreated) {
+    MP_RETURN_ON_FAIL(
+        graph.AddPacketToInputStream(
+            "input", ::mediapipe::Adopt(new std::string(*request)).At(currentTimestamp)),
+        "failed to deserialize",
+        StatusCode::MEDIAPIPE_GRAPH_ADD_PACKET_INPUT_STREAM);
+    numberOfPacketsCreated = 1;
+    return StatusCode::OK;
+}
+
+Status validateSubsequentRequestImpl(
+    const std::string& request,
+    const std::string& endpointName,
+    const std::string& endpointVersion,
+    stream_types_mapping_t& inputTypes) {
+    return StatusCode::OK;
```

Comment:
Is this complete implementation? Don't we need to do anything here?

Reply:
Yes, it is complete implementation, the validation happens before reaching MP graph (in http parser) and will exist in MP calculator (for chat completion parameters)

Reply:
However, having this call is required because gRPC implements it...